Ensure virtualenv uses python2

nikhilxb opened the pull request:
- Explicitly make sure to use a version of virtualenv belonging to the
site-packages of a python2 distribution.
- Immediately upgrade pip because of “Cache entry deserialization
failed, entry ignored” bug on Mac OSX for default pip 9.0.1 install

nikhilxb committed:
Ensure virtualenv uses python2

- Explicitly make sure to use a version of virtualenv belonging to the
site-packages of a python2 distribution.
- Immediately upgrade pip because of “Cache entry deserialization
failed, entry ignored” bug on Mac OSX for default pip 9.0.1 install

percyliang commented:
Did you test whether this works on Ubuntu?

nikhilxb commented:
Yeah I just saw the test crashed ... I’ll work on fixing that now l. It
turns out that getting virtualenvs to work in a cross-platform way is
non-trivial.

I’m wondering what preconditions we can impose: if we say that your default
python has to be 2, then I can just add a check to output an error with the
tips I posted to GitHub. You see something like this in the
CodaLab-worksheets setup.sh

On Sun, Oct 7, 2018 at 9:49 PM Percy Liang <notifications@github.com> wrote:

> Did you test whether this works on Ubuntu?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/codalab/codalab-cli/pull/978#issuecomment-427721270>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AQD_YnWIN3VGMP6D-W5_3oVCAfCRcX7fks5uitlxgaJpZM4XMPse>
> .
>
-- 
*Nikhil Bhattasali*
Stanford University
B.S. Symbolic Systems (AI)
M.S. Computer Science (AI + Bio-Robotics)
nikhilxb@stanford.edu | linkedin.com/in/nikhilbhattasali

percyliang commented:
A hacky thing is to just branch on Linux versus MacOS...

On Sun, Oct 7, 2018 at 9:54 PM Nikhil Bhattasali <notifications@github.com>
wrote:

> Yeah I just saw the test crashed ... I’ll work on fixing that now l. It
> turns out that getting virtualenvs to work in a cross-platform way is
> non-trivial.
>
> I’m wondering what preconditions we can impose: if we say that your default
> python has to be 2, then I can just add a check to output an error with the
> tips I posted to GitHub. You see something like this in the
> CodaLab-worksheets setup.sh
>
> On Sun, Oct 7, 2018 at 9:49 PM Percy Liang <notifications@github.com>
> wrote:
>
> > Did you test whether this works on Ubuntu?
> >
> > —
> > You are receiving this because you authored the thread.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/codalab/codalab-cli/pull/978#issuecomment-427721270
> >,
> > or mute the thread
> > <
> https://github.com/notifications/unsubscribe-auth/AQD_YnWIN3VGMP6D-W5_3oVCAfCRcX7fks5uitlxgaJpZM4XMPse
> >
> > .
> >
> --
> *Nikhil Bhattasali*
> Stanford University
> B.S. Symbolic Systems (AI)
> M.S. Computer Science (AI + Bio-Robotics)
> nikhilxb@stanford.edu | linkedin.com/in/nikhilbhattasali
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/codalab/codalab-cli/pull/978#issuecomment-427721733>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAakuDQ4qfIaT0T3NSUnMBap7R0QH_f5ks5uitqSgaJpZM4XMPse>
> .
>

nikhilxb committed:
Rollback and improved error msg

- Per discussion with the team, since its so hard to get virtualenv
working for Ubuntu and Mac, on standard and conda distributions, we
will ensure the developer knows that they must use a Python 2
virtualenv or create a conda env with a Python 2 virtualenv.

bkgoksel reviewed (approved):
LGTM in this state.
